Submission

KSKundan opened the pull request.

KSKundan committed:
P-Box

KSKundan committed:
R-Box

KSKundan committed:
P-Box and R-Box scheduling

KSKundan committed:
restarter is rbox and restart-service.js is p-box

KSKundan committed:
improved R-Box

This file contains the code for R-box and it improved all reviewed point.

KSKundan committed:
R-Box

KSKundan committed:
P-Box 

Executenow, schedulenow and cancel endpoint implimented

KSKundan committed:
Add files via upload

KSKundan committed:
Create readME

KSKundan committed:
Please review schedule and cancel end point

KSKundan committed:
Create readME

KSKundan committed:
All module done

all endpoint executenow, schedulenow, cancel and retrive has been completed. Please review these code.

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete readME

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+const express = require('express');
+var scheduler = require('node-cron');
```
const /var

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+            }, (error, response, body) => {
+
+                if (error) {
+                    console.log(error);
```
make a single log

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+        //////////////////////////////////////////////////Pbox section////////////////
+
+        else if (requireData.destination == 'PBOX') {
+            console.log("Entered ip ", requireData.iP);
```
remove log

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+            console.log("Entered ip ", requireData.iP);
+            var child = cp.spawn('pm2',['restart','ecosystem.config.js','--env', 'production'], {cwd:"/home/padmin/build/pm2_config"});
+            child.stdout.on('data', data => {
+                console.log('data', data.toString());
```
remove log

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+                console.log('error', error);
+            });
+            console.log("P-Box is executed successfully !!!");
+            res.status(200).send("P-Box is executed successfully !!!");
```
send res when execution is done

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+
+app.post('/schedulenow', (req, res) => {
+   
+    var destination = (req.body.type).toUpperCase();
```
add obj value check

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+            jobPbox = scheduler.schedule(dateValidateP, () => {
+                var child = cp.spawn('pm2', ['restart', 'ecosystem.config.js', '--env', 'production'], { cwd: "/home/padmin/build/pm2_config" });
+                child.stdout.on('data', data => {
+                    console.log('data', data.toString());
```
remove log

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+        if (JobDetails.Pbox_JobID != null) {
+            dateValidateP = (minut + ' ' + hour + ' ' + date + ' ' + month + ' ' + day);
+            jobPbox = scheduler.schedule(dateValidateP, () => {
+                var child = cp.spawn('pm2', ['restart', 'ecosystem.config.js', '--env', 'production'], { cwd: "/home/padmin/build/pm2_config" });
```
add path in a config file

nihal211095 commented on PboxRbox/restart-service.js:
```diff
@@ -0,0 +1,298 @@
+                });
+            });
+            schedulePboxExecuteStatu = jobPbox;
+            console.log('Reboot request for P-Box on IP'+schedulePoxIP+' is accepted');
```
res on close or exit of the spawn

KSKundan committed:
Create readME

KSKundan committed:
Add files via upload

KSKundan committed:
new P-BOX

KSKundan committed:
Delete restart-service.js

KSKundan committed:
P-Box

KSKundan committed:
Delete restarter.js

KSKundan committed:
Delete readme.MD

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete readME

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete readME

KSKundan committed:
Delete restarter.js

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete readME

KSKundan committed:
Create readme.MD

KSKundan committed:
Final upload 

P-Box is restart-service.js and R-Box is restarter.js
All the data like cmd, port, ip are imported from config.js

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Add files via upload

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete restarter.js

KSKundan committed:
Delete config.js

KSKundan committed:
Delete readme.MD

KSKundan committed:
Create readME

KSKundan committed:
final upload 

Restart final upload

KSKundan committed:
Delete restarter.js

KSKundan committed:
Delete restart-service.js

KSKundan committed:
Delete config.js

KSKundan committed:
Add files via upload

KSKundan committed:
Update readME

KSKundan committed:
Update readME

KSKundan committed:
Create readme.MD

KSKundan committed:
Add files via upload

KSKundan committed:
Create readme.MD

KSKundan committed:
restarter for pBox and rBox

Restart-service.js is pbox file and restarter.js is rBox file

KSKundan committed:
Create readME

KSKundan committed:
restarter modified file
